opennlp-tools/src/main/java/opennlp/tools/ml/model/ComparablePredicate.java

```diff
@@ -34,20 +36,54 @@ public ComparablePredicate(String n, int[] ocs, double[] ps) {
   }
 
   public int compareTo(ComparablePredicate cp) {
-    int smallerLength = outcomes.length > cp.outcomes.length ?
-        cp.outcomes.length : outcomes.length;
+    int smallerLength = Math.min(outcomes.length, cp.outcomes.length);
 
     for (int i = 0; i < smallerLength; i++) {
-      if (outcomes[i] < cp.outcomes[i]) return -1;
-      else if (outcomes[i] > cp.outcomes[i]) return 1;
+      int compareOutcomes = Integer.compare(outcomes[i], cp.outcomes[i]);
+      if (compareOutcomes != 0) {
+        return compareOutcomes;
+      }
     }
 
-    if (outcomes.length < cp.outcomes.length) return -1;
-    else if (outcomes.length > cp.outcomes.length) return 1;
+    int compareOutcomesLength = Integer.compare(outcomes.length, cp.outcomes.length);
+    if (compareOutcomesLength != 0) {
+      return compareOutcomesLength;
+    }
 
     return 0;
   }
 
+  @Override
+  public int hashCode() {
+    final int prime = 31;
```

Comment:
We use a different style for hashCode in this project, Java now contains via Objects utils to do this nicer.